jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt

```diff
@@ -61,73 +24,19 @@ class JavaLambdaBuilder : LambdaBuilder() {
-        val compileScope = compilerManager.createModulesCompileScope(arrayOf(module), true, true)
+        val baseDir = getBaseDirectory(module).path
+        val customTemplate = FileUtil.createTempFile("template", ".yaml", true)
+        SamTemplateUtils.writeDummySamTemplate(customTemplate, runtime, baseDir, handler, envVars)
 
-        compilerManager.make(compileScope) { aborted, errors, _, context ->
-            if (!aborted && errors == 0) {
-                try {
-                    copyLibraries(module, buildDir)
-                    copyClasses(module, buildDir)
-                    future.complete(BuiltLambda(null, buildDir.toPath()))
-                } catch (e: Exception) {
-                    future.completeExceptionally(RuntimeException(message("lambda.package.zip_fail"), e))
-                }
-            } else if (aborted) {
-                future.completeExceptionally(RuntimeException(message("lambda.package.compilation_aborted")))
-            } else {
-                val errorMessages = context.getMessages(CompilerMessageCategory.ERROR).joinToString("\n")
-                future.completeExceptionally(
-                    RuntimeException(
-                        message(
-                            "lambda.package.compilation_errors",
-                            errorMessages
-                        )
-                    )
-                )
-            }
-        }
-        return future
-    }
-
-    private fun copyLibraries(module: Module, buildDir: File) {
-        val libDir = File(buildDir, "lib")
-        productionRuntimeEntries(module)
-            .librariesOnly()
-            .pathsList.pathList
-            .map { File(it) }
-            .filter { it.exists() }
-            .forEach { copyFileOrDir(it, libDir) }
-    }
-
-    private fun copyClasses(module: Module, buildDir: File) {
-        productionRuntimeEntries(module)
-            .withoutLibraries()
-            .classes()
-            .pathsList.pathList
-            .map { File(it) }
-            .filter { it.exists() }
-            .forEach { copyFileOrDir(it, buildDir) }
+        return buildLambdaFromTemplate(module, customTemplate.toPath(), "Function", samOptions)
     }
 
-    private fun productionRuntimeEntries(module: Module) = OrderEnumerator.orderEntries(module)
-        .recursively()
-        .productionOnly()
-        .runtimeOnly()
-        .withoutSdk()
-
-    private fun copyFileOrDir(source: File, dest: File) {
-        if (source.isDirectory) {
-            FileUtil.copyDir(source, dest, false)
-        } else {
-            FileUtil.copy(source, File(dest, source.name))
+    private fun getBaseDirectory(module: Module): VirtualFile {
+        val rootManager = ModuleRootManager.getInstance(module)
+        val contentRoots = rootManager.contentRoots
+        return when (contentRoots.size) {
+            1 -> contentRoots[0]
+            else -> throw IllegalArgumentException("Module can only have 1 content root")
```

Comment:
Do we need to provide messaging to customers, or is multi-content root rare enough?

Reply:
It should be rare in Java land, but ill add a message

Reply:
Added